skimage/restoration/tests/test_denoise.py

```diff
@@ -308,5 +308,21 @@ def test_no_denoising_for_small_h():
     assert np.allclose(denoised, img)
 
 
+def test_wavelet_denoising():
+    for img in [astro_gray, astro]:
+        img = img.copy() + 0.1 * np.random.randn(*(img.shape))
+        img = np.clip(img, 0, 1)
+        # less energy in signal
+        assert restoration.denoise_wavelet(img, sigma=0.3).sum() <= img.sum()
+
+        # test changing noise_std (higher threshold, so less energy in signal)
+        assert (restoration.denoise_wavelet(img, sigma=0.2).sum() <=
+                restoration.denoise_wavelet(img, sigma=0.1).sum())
+
+        # This works for this particular image
+        assert (np.sum(restoration.denoise_wavelet(img, wavelet='db2')) <
```

Comment:
I think a good test would be comparing the original image vs. noisy with `compare_ssim` and the original vs. denoised also with `compare_ssim`. The denoised comparison should be better.